File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        
        ans = [] 
        if not root: return ans 
        
        def dfs(node,dep):
            
            if len(ans) == dep:
                ans.append([])
            
            ans[dep].append(node.val)
            for child in node.children: 
                dfs(child,dep+1)
        
        dfs(root,0)
        return ans 
    
    
        

File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<vector<int>> ans ; 
    vector<vector<int>> levelOrder(Node* root) {
        
        dfs(root,0) ; 
        return ans ; 

}
    
void dfs(Node* root  , int dep){

    if(root == nullptr) return ; 
    if(dep == ans.size()) ans.emplace_back() ; 
    ans[dep].push_back(root->val); 
    for ( Node* child : root->children) 
        dfs(child , dep +1 ) ; 
}

};